### DISCUSSION TITLE
Deploy script adds disallowed property name to network in docker-stack.yml

### DISCUSSION THREAD

**Discussion by andretheronsa** [fastapi/full-stack-fastapi-template#1822] [Questions] [answered]
When running the _deploy.sh_ script `docker stack deploy -c ...` fails with:
> networks.traefik-public Additional property name is not allowed

The following gets added to the docker-stack.yml by `docker-compose -f docker-compose.yml config > docker-stack.yml`

`networks:
traefik-public:
    external: true
    name: traefik-public`

This goes for docker-compose file version 3.3 and 3.6.

If I remove the `name: traefik-public` line everything works fine. I am not that experienced with docker networks but why is config adding a property that docker stack deploy does not allow?

----------------------------------------------
Ubuntu 20.04
Docker version 20.10.2, build 2291f61
docker-compose version 1.27.4, build 40524192
docker-py version: 4.3.1
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019

**Comment by ttamg** [comment, marked as answer]
I also have the same issue.  

Using:
* Ubuntu 20.04
* Docker 19.03.13
* docker-compose 1.27.4

Simple fix was to move the version up to 3.6 in the `docker-compose.yml` and that then enabled it to spin up the stack.
